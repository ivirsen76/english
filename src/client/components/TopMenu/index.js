import React from 'react'
import PropTypes from 'prop-types'
import { Dropdown, Menu, Modal } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'
import LoginForm from 'components/LoginForm'

const TopMenu = ({ isLoggedIn, user }) =>
    <Menu inverted>
        <MenuLink to="/" onlyActiveOnIndex>
            Home
        </MenuLink>
        <MenuLink to="/features">Features</MenuLink>
        <Menu.Item>Card base</Menu.Item>
        <Dropdown item text="Helpers">
            <Dropdown.Menu>
                <Dropdown.Item>Матрица Simple</Dropdown.Item>
                <Dropdown.Item>Матрица Continuous</Dropdown.Item>
                <Dropdown.Item>Выделение слов</Dropdown.Item>
                <Dropdown.Item>Числа на слух</Dropdown.Item>
            </Dropdown.Menu>
        </Dropdown>
        {isLoggedIn
            ? <Menu.Menu position="right">
                  <MenuLink to="/user">Dashboard</MenuLink>
                  <Dropdown
                      item
                      trigger={
                          <span>
                              <i className="icon-user" /> {user.email}
                          </span>
                      }
                  >
                      <Dropdown.Menu>
                          <Dropdown.Item>Change password</Dropdown.Item>
                          <MenuLink to="/logout">Log out</MenuLink>
                      </Dropdown.Menu>
                  </Dropdown>
              </Menu.Menu>
            : <Menu.Menu position="right">
                  <Modal size="small" closeIcon trigger={<a className="item">Log in</a>}>
                      <Modal.Header>Log in</Modal.Header>
                      <Modal.Content>
                          <LoginForm />
                      </Modal.Content>
                  </Modal>
              </Menu.Menu>}
    </Menu>

TopMenu.propTypes = {
    isLoggedIn: PropTypes.bool,
    user: PropTypes.object,
}

export default TopMenu
