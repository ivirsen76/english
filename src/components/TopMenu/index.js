import React, { PropTypes } from 'react'
import { Dropdown, Menu, Modal } from 'semantic-ui-react'
import MenuLink from 'components/MenuLink'
import LoginForm from 'components/LoginForm'


const TopMenu = ({ isLoggedIn, user }) => (
    <Menu inverted>
        <MenuLink to="/" onlyActiveOnIndex>Главная</MenuLink>
        <MenuLink to="/features">Возможности</MenuLink>
        <Menu.Item>Базы слов</Menu.Item>
        <Dropdown item text="Помощники">
            <Dropdown.Menu>
                <Dropdown.Item>Матрица Simple</Dropdown.Item>
                <Dropdown.Item>Матрица Continuous</Dropdown.Item>
                <Dropdown.Item>Выделение слов</Dropdown.Item>
                <Dropdown.Item>Числа на слух</Dropdown.Item>
            </Dropdown.Menu>
        </Dropdown>
        {isLoggedIn ? (
            <Menu.Menu position="right">
                <MenuLink to="/user">Dashboard</MenuLink>
                <Dropdown item trigger={<span><i className="icon-user" /> {user.email}</span>}>
                    <Dropdown.Menu>
                        <Dropdown.Item>Сменить пароль</Dropdown.Item>
                        <MenuLink to="/logout">Выйти</MenuLink>
                    </Dropdown.Menu>
                </Dropdown>
            </Menu.Menu>
        ) : (
            <Menu.Menu position="right">
                <Modal
                    size="small"
                    closeIcon
                    trigger={<a className="item">Войти</a>}
                >
                    <Modal.Header>Войти</Modal.Header>
                    <Modal.Content>
                        <LoginForm />
                    </Modal.Content>
                </Modal>
            </Menu.Menu>
        )}
    </Menu>
)

TopMenu.propTypes = {
    isLoggedIn: PropTypes.bool,
    user: PropTypes.object,
}

export default TopMenu
