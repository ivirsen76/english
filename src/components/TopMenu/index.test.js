/* global describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import { toJson } from '@ieremeev/jest-setup';
import TopMenu from './index';


describe('<TopMenu />', () => {
    it('Should render menu', () => {
        const wrapper = shallow(<TopMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
