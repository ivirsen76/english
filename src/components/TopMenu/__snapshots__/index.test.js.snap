// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopMenu /> Should render menu 1`] = `
<Menu
  inverted={true}
>
  <withProps(Link)
    onlyActiveOnIndex={true}
    to="/"
  >
    Главная
  </withProps(Link)>
  <withProps(Link)
    to="/features"
  >
    Возможности
  </withProps(Link)>
  <MenuItem>
    Базы слов
  </MenuItem>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    item={true}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    text="Помощники"
  >
    <DropdownMenu>
      <DropdownItem>
        Матрица Simple
      </DropdownItem>
      <DropdownItem>
        Матрица Continuous
      </DropdownItem>
      <DropdownItem>
        Выделение слов
      </DropdownItem>
      <DropdownItem>
        Числа на слух
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <MenuMenu
    position="right"
  >
    <Modal
      closeIcon={true}
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      size="small"
      trigger={Object {}}
    >
      <ModalHeader>
        Войти
      </ModalHeader>
      <ModalContent>
        <ReduxForm
          onSubmit={[Function]}
        />
      </ModalContent>
    </Modal>
  </MenuMenu>
</Menu>
`;

exports[`<TopMenu /> Should render menu for logged user 1`] = `
<Menu
  inverted={true}
>
  <withProps(Link)
    onlyActiveOnIndex={true}
    to="/"
  >
    Главная
  </withProps(Link)>
  <withProps(Link)
    to="/features"
  >
    Возможности
  </withProps(Link)>
  <MenuItem>
    Базы слов
  </MenuItem>
  <Dropdown
    additionLabel="Add "
    additionPosition="top"
    closeOnBlur={true}
    icon="dropdown"
    item={true}
    noResultsMessage="No results found."
    openOnFocus={true}
    renderLabel={[Function]}
    selectOnBlur={true}
    text="Помощники"
  >
    <DropdownMenu>
      <DropdownItem>
        Матрица Simple
      </DropdownItem>
      <DropdownItem>
        Матрица Continuous
      </DropdownItem>
      <DropdownItem>
        Выделение слов
      </DropdownItem>
      <DropdownItem>
        Числа на слух
      </DropdownItem>
    </DropdownMenu>
  </Dropdown>
  <MenuMenu
    position="right"
  >
    <withProps(Link)
      to="/user"
    >
      Dashboard
    </withProps(Link)>
    <Dropdown
      additionLabel="Add "
      additionPosition="top"
      closeOnBlur={true}
      icon="dropdown"
      item={true}
      noResultsMessage="No results found."
      openOnFocus={true}
      renderLabel={[Function]}
      selectOnBlur={true}
      trigger={Object {}}
    >
      <DropdownMenu>
        <DropdownItem>
          Сменить пароль
        </DropdownItem>
        <DropdownItem>
          Выйти
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </MenuMenu>
</Menu>
`;
